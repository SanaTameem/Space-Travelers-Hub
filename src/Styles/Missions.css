.missionsPage {
  background-color: #f5f5f5;
  padding: 50px 0;

  /* background-color: blue; */
  width: 100%;
}

.missionTable {
  width: 100%;
  border-collapse: collapse;
  border-spacing: 0;

  /* background-color: yellow; */
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  border: 1px solid #b1b1b1;
}

h3 {
  margin: 0;
  padding: 8px;
  font-size: 1.2em;
  font-weight: 600;
  color: #333;
  border: 1px solid #b1b1b1;
  text-align: left;
}

.cell {
  border: 1px solid #b1b1b1;

  /* background-color: red; */
  height: 150px;
  overflow: hidden;
}

.missionTable .column:nth-child(even) .cell {
  background-color: #f5f5f5; /* Even row color */
}

.missionTable .column:nth-child(odd) .cell {
  background-color: #fff; /* Odd row color */
}
