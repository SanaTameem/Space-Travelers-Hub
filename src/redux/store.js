import { configureStore } from '@reduxjs/toolkit';
import rocketReducer from './rockets/rocketsSlice';
import missionReducer from './missions/missionsSlice';
import dragonReducer from './dragons/dragonsSlice';

const store = configureStore({
  reducer: {
    rocket: rocketReducer,
    mission: missionReducer,
    dragon: dragonReducer,
  },
});

export default store;
