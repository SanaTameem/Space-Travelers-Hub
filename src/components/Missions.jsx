import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchMissions,
  joinMission,
  unjoinMission,
} from '../redux/missions/missionsSlice';
import '../Styles/Missions.css';

const Mission = () => {
  const missions = useSelector((state) => state.mission.missions);
  const dispatch = useDispatch();

  useEffect(() => {
    if (missions.length === 0) {
      dispatch(fetchMissions());
    }
  }, [dispatch]);

  return (
    <section className="missionsPage">
      <div className="missionTable">
        <div className="column">
          <h3>Missions</h3>
          {missions.length > 0 ? (
            missions.map((mission) => (
              <div key={mission.id} className="cell">
                {mission.mission_name}
              </div>
            ))
          ) : (
            <div>No missions available</div>
          )}
        </div>

        <div className="column">
          <h3>Descriptions</h3>
          {missions.length > 0 ? (
            missions.map((mission) => (
              <div key={mission.id} className="cell">
                {mission.description}
              </div>
            ))
          ) : (
            <div>No descriptions available</div>
          )}
        </div>

        <div className="column">
          <h3>Status</h3>
          {missions.map((mission) => (
            <div key={mission.id} className="cell">
              {mission.reserved ? (
                <span className="activeMemberBadge">Active Member</span>
              ) : (
                <span>NOT A MEMBER</span>
              )}
            </div>
          ))}
        </div>

        <div className="column">
          <h3>Actions</h3>
          {missions.map((mission) => (
            <div key={mission.id} className="cell">
              {mission.reserved ? (
                <button
                  type="submit"
                  onClick={() => dispatch(unjoinMission(mission.id))}
                >
                  Leave Mission
                </button>
              ) : (
                <button
                  type="submit"
                  onClick={() => dispatch(joinMission(mission.id))}
                >
                  Join Mission
                </button>
              )}
            </div>
          ))}
        </div>
      </div>
    </section>
  );
};

export default Mission;
