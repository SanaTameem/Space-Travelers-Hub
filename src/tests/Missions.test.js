import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import Missions from '../components/Missions';

describe('Missions', () => {
  test('Missions are rendered correctly', () => {
    const myMissions = render(
      <Provider store={store}>
        <Missions />
      </Provider>,
    );
    expect(myMissions).toMatchSnapshot();
  });
});
