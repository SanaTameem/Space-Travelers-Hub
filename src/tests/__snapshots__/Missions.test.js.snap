// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Missions Missions are rendered correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="missionsPage"
      >
        <div
          class="missionTable"
        >
          <div
            class="column"
          >
            <h3>
              Missions
            </h3>
            <div>
              No missions available
            </div>
          </div>
          <div
            class="column"
          >
            <h3>
              Descriptions
            </h3>
            <div>
              No descriptions available
            </div>
          </div>
          <div
            class="column"
          >
            <h3>
              Status
            </h3>
          </div>
          <div
            class="column"
          >
            <h3>
              Actions
            </h3>
          </div>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="missionsPage"
    >
      <div
        class="missionTable"
      >
        <div
          class="column"
        >
          <h3>
            Missions
          </h3>
          <div>
            No missions available
          </div>
        </div>
        <div
          class="column"
        >
          <h3>
            Descriptions
          </h3>
          <div>
            No descriptions available
          </div>
        </div>
        <div
          class="column"
        >
          <h3>
            Status
          </h3>
        </div>
        <div
          class="column"
        >
          <h3>
            Actions
          </h3>
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
